### DISCUSSION TITLE
Feature: highlight active region in `vertico-buffer-mode`'s minibuffer

### DISCUSSION THREAD

**Discussion by hab25** [minad/vertico#413] [Ideas]
Reproduced with `emacs -Q`:
- Make sure vertico is in the `load-path`, then eval the following
```
(progn
  (require 'vertico-buffer)
  (vertico-mode)
  (vertico-buffer-mode))
```
- Do `M-x`, type some word and try selecting some amount of chars e.g., with `C-S-<left>`.
- See that the region is not highlighted (but is still there; e.g., try `<backspace>`)

Note: the highligting starts working again after one mouse-clicks on the vertico buffer.

Vertico version 1.4 from GNU ELPA.

**Comment by minad** [comment]
Thanks for the report. I've also observed this before. I've migrated the issue to an "Ideas" discussion since it is essentially a missing feature.

For some reason `transient-mark-mode` doesn't highlight the active region in another window. In its current implementation, `vertico-buffer-mode` duplicates the minibuffer. The minibuffer is still shown in the miniwindow (but hidden) and also in the separate large window. To demonstrate this, you can set `vertico-buffer-hide-prompt` to nil, such that the original minibuffer window content stays visible. 

I have not found a simple solution so far to also replicate the marked region in the separate window. I think one somehow has to copy the `transient-mark-mode` state to the other window. Patches implementing this feature are welcome.

**Reply by hab25** [reply]
Thank you for the maintenance. Does Vertico provide an easy way of having Vertico UI to display at the top of the frame [^1] while having this feature of highlighting the active region?

[^1]: like  `vertico-buffer` with ``(setopt vertico-buffer-display-action `(display-buffer-in-side-window (window-height . ,(+ 3 vertico-count)) (side . top)))``

I tried the following alternatives, but they did not meet my needs: `vertico-posframe` also doesn't have this feature (https://github.com/tumashu/vertico-posframe/issues/31), `mini-frame` suffers from two dealbreaker bugs/annoyances: https://github.com/muffinmad/emacs-mini-frame/issues/67 and https://github.com/muffinmad/emacs-mini-frame/issues/25

**Reply by hab25** [reply]
Many popular desktop programs, such as Chromium, GNOME, and VSCode, have their "minibuffer" at the top of the frame by default; I expect that Emacs newcomers would appreciate `vertico-buffer-mode` being a more featureful experience.

**Reply by minad** [reply]
Not as far as I know. Our best bet is to improve `vertico-buffer-mode` accordingly.

**Reply by minad** [reply]
Implemented in https://github.com/minad/vertico/commit/996d1223d2be41023c0331f00176a633a84a7d1d.
